compiler/src/iree/compiler/Preprocessing/Common/PadToIntrinsics.cpp

```diff
@@ -198,15 +504,26 @@ struct PadToIntrinsicsPass final
           return GPUMatmulShapeType{mSize, nSize, kSize, aType, bType, cType};
         });
 
-    SmallVector<linalg::Conv2DNhwcHwcfOp> targetOps;
-    rootOp->walk(
-        [&](linalg::Conv2DNhwcHwcfOp convOp) { targetOps.push_back(convOp); });
+    SmallVector<linalg::LinalgOp> targetOps;
+    rootOp->walk([&](linalg::LinalgOp linalgOp) {
+      if (isa<linalg::Conv2DNhwcHwcfOp, linalg::BatchMatmulOp, linalg::MatmulOp,
+              linalg::MatmulTransposeBOp, linalg::GenericOp>(
+              linalgOp.getOperation()))
+        targetOps.push_back(linalgOp);
+    });
 
     IRRewriter rewriter(context);
-    for (linalg::Conv2DNhwcHwcfOp convOp :
-         llvm::make_early_inc_range(targetOps)) {
-      rewriter.setInsertionPoint(convOp);
-      padConvOp(rewriter, convOp, intrinsics);
+    for (auto linalgOp : llvm::make_early_inc_range(targetOps)) {
```

Comment:
Ok, my bad here (probably dont know who committed the original sin), but better to walk the program and collect the operations to transform and then do the transformation (to avoid doing `make_early_inc_range`. Will also help prune `GenericOp` that are not contraction ops during the walk.

Reply:
Sorry Mahesh, I am not sure if I fully understand. Aren't we already doing a rootOp->walk? Should we update the rootOp->walk to only add `GenericOp` if `linalg::isaContractionOpInterface` is true and replace the `make_early_inc_range` with regular loop? 

Also is there anything wrong with `make_early_inc_range`? haha

Reply:
early inc range allows you to iterate over (intrusive) link lists whose nodes get invalidated by the current iteration. We can use a regular for loop here, I missed this when cleaning up this code.

Reply:
I removed the make_early_inc_range, and only pushed in ops that matches our option into the worklist.